File: website/Logic.js
class Logic{

    boardWidth;
    boardHeight;
    currentPostion;
    x;
    y;
    finalPosition;


    constructor(width,height)
    {
        this.boardWidth = width;
        this.boardHeight = height;

    }

    setPosition(currentPostion)
    {
        this.currentPostion = currentPostion;

    }

    async getFinalPosition()
    {
        let x = await this.getX();
        let y = await this.getY();
        return [x,y]

    }

    async getX()
    {
        let numDevideTen =Math.floor(this.currentPostion/10);
        if(this.isInteger(this.currentPostion/10))
        {
            if(this.isPrime(numDevideTen))
            {
              return this.boardX(1);
            }
            else
            {
               return this.boardX(10);
            }
        }

        let num =this.currentPostion/10;
        let decimal = Math.round(((num -Math.floor(this.currentPostion/10))*10) * 10) / 10;
        let decimalPart = decimal;
       if(this.isPrime(numDevideTen))
       {
         return this.boardX(decimalPart);
       }
       else
       {
         return this.boardX(11 - decimalPart);
       }
    }

    boardX(number)
	{
        
        return (this.boardWidth/10)*number - (this.boardWidth/20) 
	}

    async getY()
    {
        if(this.isInteger(this.currentPostion/10))
		{
			return this.boardY(this.currentPostion/10);
		}
		
		return this.boardY(Math.floor(this.currentPostion/10 + 1));
    }
	
	boardY(number)
	{
		return this.boardHeight - ((Math.floor(this.boardHeight/10))*number - (Math.floor(this.boardWidth/30)));
	}
	
	isPrime(number)
	{
		if(number %2 == 0)
			return true;
		else
			return false;
	}
	
	isInteger(number)
	{
		if(number%1.0 === 0.0)
			return true;
		else
			return false;
	
	}
}

File: src/Main/Logic.java
package Main;

public class Logic {

	private int position;
	private int[] finalPosition;
	
	Logic()
	{
		
	}
	
	void setPosition(int position)
	{
		this.position = position;
		setFinalPosition(new int[] {getX(),getY()});
	}
	
	public int getX()
	{
		   int numDevideTen = position/10;
		    if(isInteger((double) position/10))
		    {
		        if(isPrime(numDevideTen))
		        {
		          return 10;
		        }
		        else
		        {
		           return BoardAndXY.boardX-50;
		        }
		    }

		    double num =(double) position/10;
		    double decimal = (double)Math.round(((num - (int)(position/10))*10) * 100000d) / 100000d;
		    int decimalPart = (int) decimal;
		   if(isPrime(numDevideTen))
		   {
		     return boardX(decimalPart);
		   }
		   else
		   {
		     return boardX(11 - decimalPart);
		   }
	}
	
	private int boardX(int number)
	{
		return (number-1)*60 + 10;
	}
	
	public int getY()
	{
		if(isInteger((double) position/10))
		{
			return boardY(position/10);
		}
		
		return boardY(position/10 + 1);
	}
	
	private int boardY(int number)
	{
		return BoardAndXY.boardY - (number * 60);
	}
	
	boolean isPrime(int number)
	{
		if(number %2 == 0)
			return true;
		else
			return false;
	}
	
	boolean isInteger(double number)
	{
		if(number%1.0 == 0.0)
			return true;
		else
			return false;
	
	}

	public int[] getFinalPosition() {
		return finalPosition;
	}

	public void setFinalPosition(int[] finalPosition) {
		this.finalPosition = finalPosition;
	}
}
